src/codegen/riscv64/constants-riscv64.h

```diff
@@ -978,6 +978,36 @@ class InstructionGetters : public T {
     return imm6 << 26 >> 26;
   }
 
+  inline int RvcImm6UValue() const {
+    DCHECK(this->IsShortInstruction());
+    // | funct3 | nzimm[17] | rs1/rd | nzimm[16:12] | opcode |
+    //  15         12                 6            2        
+    int32_t imm6 = this->RvcImm6Value();
+    DCHECK(imm6 != 0);
+    return (uint32_t)imm6 << 12 >> 12;
```

Comment:
Shouldn't it be shifting by (32-6) = 24 bits?

Reply:
I found the code is difficult to understand. I just want to return the same result as `Imm20UValue()`.

Please review the new code.

```
  inline int RvcImm6UValue() const {
    DCHECK(this->IsShortInstruction());
    // | funct3 | nzimm[17] | rs1/rd | nzimm[16:12] | opcode |
    //  15         12                 6            2        
    int32_t imm6 = this->RvcImm6Value();
    DCHECK(imm6 != 0);
    return imm6 & 0xFFFFF;
  }
```

Reply:
I agree that using AND to zero-extend is easier to understand and less error-prone than using unsigned shift-left and right. The name of `RvcImm6UValue` and `RvcImm6Value` imply you are extracting an unsigned and signed 6-bit immediate. If you want to extract 20-bit immediate (even though the lower 14 bits are automatically appended as 0s), the naming should reflect that, e.g., `RvcImm20Value`  and `RvcImm20UValue`.

Reply:
I thought the "U" here means "Upper". Because the immediate in `Imm20UValue()` is also sign-extended which is used by LUI instruction.

> LUI (load upper immediate) uses the same opcode as RV32I. LUI places the 32-bit U-immediateinto registerrd, filling in the lowest 12 bits with zeros.  The 32-bit result is sign-extended to 64 bits.

And the number 6 means 6-bit in the instruction, not the length of result, just like `Imm12Value()` and `Imm20UValue()`. So I prefer to keep them.

Here is `0xFFFFF` because I found that the result of toolchain's disasm is 20-bit length.

Reply:
I see. If this is just to extract 6-bit signed value from the instruction, no need to add `0xFFFFF`. Since `& 0xFFFFF` is only for `disasm`, it is better to do the `& 0xFFFF` part in `Decoder::PrintRvcImm6U`, so that the meaning of `RvcImm6UValue` is clear.

Also, if `& 0xFFFFF` is moved out of `RvcImm6UValue`, then `RvcImm6UValue` is the same as `RvcImm6Value`, then we can  remove `RcvImm6UValue` altogether.

Reply:
Good idea. I will do it.